datafusion/spark/src/function/string/ascii.rs

```diff
@@ -0,0 +1,210 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::array::{ArrayAccessor, ArrayIter, ArrayRef, AsArray, Int32Array};
+use arrow::datatypes::DataType;
+use arrow::error::ArrowError;
+use datafusion_common::{internal_err, plan_err, Result};
+use datafusion_expr::{ColumnarValue, Documentation};
+use datafusion_expr::{ScalarFunctionArgs, ScalarUDFImpl, Signature, Volatility};
+use datafusion_functions::utils::make_scalar_function;
+use datafusion_macros::user_doc;
+use std::any::Any;
+use std::sync::Arc;
+
+#[user_doc(
+    doc_section(label = "Spark String Functions"),
+    description = "Returns the ASCII code point of the first character of string.",
+    syntax_example = "ascii(str)",
+    sql_example = r#"```sql
+> select ascii('abc');
++--------------------+
+| ascii(abc)         |
++--------------------+
+| 97                 |
++--------------------+
+> select ascii('🚀');
++-------------------+
+| ascii(🚀)         |
++-------------------+
+| 128640            |
++-------------------+
+> select ascii(2);
++----------------+
+| ascii(2)       |
++----------------+
+| 50             |
++----------------+
+> select ascii(X'44617461467573696F6E');
++--------------------------------------+
+| ascii(X'44617461467573696F6E')       |
++--------------------------------------+
+| 68                                   |
++--------------------------------------+
```

Comment:
I think this has some non trivial replication with the spark documentation -- rather than add docs directly to DataFusion's docs, what do you think about simply adding a link to the spark function? In this case, that would be: https://spark.apache.org/docs/latest/api/sql/index.html#ascii

?

That would make for less code in DataFusion and less chance the docs are incorrect / out of sync